src/vivarium_gates_mncnh/components/delivery_facility.py

```diff
@@ -50,15 +54,36 @@ def on_time_step(self, event: Event) -> None:
 
         pop = self.population_view.get(event.index)
         # Choose delivery facility type
-        birth_idx = pop.index[
+        not_partial_term = (
             pop[COLUMNS.PREGNANCY_OUTCOME] != PREGNANCY_OUTCOMES.PARTIAL_TERM_OUTCOME
-        ]
-        delivery_facility_type = self.randomness.choice(
-            birth_idx,
-            list(self.delivery_facility_probabilities.keys()),
-            p=list(self.delivery_facility_probabilities.values()),
-            additional_key="delivery_facility_type",
         )
-        pop.loc[birth_idx, COLUMNS.DELIVERY_FACILITY_TYPE] = delivery_facility_type
+        pop = pop[not_partial_term]
+        propensity = self.propensity(pop.index)
+        is_believed_preterm = pop[COLUMNS.STATED_GESTATIONAL_AGE] < 37
+
+        # home births for lower propensities, facility for higher
+        assigned_home_preterm = (
+            propensity[is_believed_preterm]
+            < self.delivery_facility_probabilities[FACILITY_CHOICE.P_HOME_PRETERM]
+        )
+        assigned_home_full_term = (
+            propensity[~is_believed_preterm]
+            < self.delivery_facility_probabilities[FACILITY_CHOICE.P_HOME_FULL_TERM]
+        )
+        assigned_home = pd.concat(
+            [assigned_home_preterm, assigned_home_full_term]
+        ).sort_index()
+        pop.loc[assigned_home, COLUMNS.DELIVERY_FACILITY_TYPE] = DELIVERY_FACILITY_TYPES.HOME
+
+        # BEmONC for lower propensities, CEmonC for higher
+        pop.loc[
+            ~assigned_home, COLUMNS.DELIVERY_FACILITY_TYPE
+        ] = DELIVERY_FACILITY_TYPES.CEmONC
+        is_bemonc = (
+            propensity < self.delivery_facility_probabilities[FACILITY_CHOICE.P_BEmONC]
```

Comment:
Are we really supposed to be using the same propensity to determine home vs in facility, and to differentiate between different facility levels? Won't all the lower propensities be given to home birth, and therefore we have not enough assigned to BEmONC.

Reply:
No - I remember reminding myself to not use the same propensity while writing the home facility portion and then forgot...To explain my check in the description: I did check the distribution of these categories and saw that no one was assigned BEmONC, but I then looked at the lowest propensity and saw that it was indeed greater than the probability of BEmONC. I thought this was acceptable but probably should have thought about it more.